lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,218 @@
+    validate(config);
+    if (!validate.errors) return [];
+    return normalizeAjvErrors(validate.errors, config, schema);
+  };
+
+  let errors;
+  before(() => {
+    errors = resolveNormalizedErrors({
+      foo: 'bar',
+      package: { incclude: ['./folder'] },
+      functions: {
+        'invalid name': {},
+        'foo': {
+          events: [
+            {
+              bar: {},
+            },
+            {
+              http: null,
+              method: 'GET',
+            },
+            {
+              http: {
+                other: 'foo',
+              },
+            },
+            {
+              http: 'gets foo',
+            },
+            {
+              http: {
+                method: 'get',
+              },
+            },
+          ],
+        },
+      },
+    });
+  });
+
+  describe('Reporting', () => {
+    it('should report error for unrecognized root property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized deep level property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.package') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for invalid function name', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.functions') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized event', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[0]") return false;
+          if (error.keyword !== 'anyOf') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for improper indentation when defining event', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[1].http") return false;
+          if (error.keyword !== 'anyOf') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error in anyOf case, where two types are possible (string and object), and object was used', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[2].http") return false;
+          if (error.keyword !== 'anyOf') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
```

Comment:
Why this test case was changed?

I believe it should stay as it was (we should just secure valid values for `path` and `method`  properties for that event, as those were added in schema to test other case)

We may also improve wording of description to:

_Should report error for unrecognized property in anyOf case, where two types are possible (string and object), and object with not supported property was used_

Reply:
This the keyword was changed from `additionalProperties` to `anyOf` because in https://github.com/serverless/serverless/pull/7335#issuecomment-661823770 you asked me to include two more cases. I added more constraints to schema and it led to error message to be like this:

```
"functions.foo.events[2].http should match schema any of: functions.foo.events[2].http should be string; Unrecognized property 'other' on 'functions.foo.events[2].http'; functions.foo.events[2].http should have required property 'path'; functions.foo.events[2].http should have required property 'method'"
```

So it is not anymore about just one unrecognized property, now it's about several submessages reported under one  `anyOf` keyword.

Reply:
> because in #7335 (comment) you asked me to include two more cases

hmm.. yes, but they should be tested with different event instances, and errors introduced with other events, should not affect error reporting for this event.

Reply:
Ideally `.functions['foo'].events[2]` should have just one error, and reporting for this error should be tested with this test case

Reply:
This makes sence. I'll move it to a separate test block